ut/columns_ut.cpp

```diff
@@ -185,6 +185,17 @@ TEST(ColumnsCase, Date_UInt16_interface) {
     ASSERT_EQ(col1->RawAt(1), 1234u);
 }
 
+TEST(ColumnsCase, Date_UInt16_construct_from_rvalue_data) {
+    auto const expected = MakeNumbers<uint16_t>();
+
+    auto data = expected;
+    auto col1 = std::make_shared<ColumnDate>(std::move(data));
+
+    ASSERT_EQ(col1->Size(), expected.size());
```

Comment:
you can use a single check instead of  for-loop here:
```C++
EXPECT_TRUE(CompareRecursive(*col1, expected));
```
(same for other places)